chain/network/src/peer_manager.rs

```diff
@@ -105,6 +129,268 @@ struct ActivePeer {
+    type Result = IbfRoutingTableExchangeMessagesResponse;
+}
+
+#[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+#[derive(MessageResponse, Debug)]
+pub enum IbfRoutingTableExchangeMessagesResponse {
+    AddIbfSetResponse {
+        seed: u64,
+        ibf_vec: Vec<IbfBox>,
+    },
+    #[cfg(feature = "adversarial")]
+    GetSeed {
+        seed: Option<u64>,
+    },
+    Empty(),
+    ProcessIbfMessageResponse {
+        ibf_msg: Option<RoutingVersion2>,
+    },
+}
+
+#[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+impl IbfRoutingTableExchangeActor {
+    pub fn exchange_routing_tables_using_ibf(
+        &self,
+        peer_id: &PeerId,
+        ibf_set: &IbfSet<SimpleEdge>,
+        ibf_level: ValidIBFLevel,
+        ibf_vec: &[IbfBox],
+        seed: u64,
+    ) -> (Vec<SimpleEdge>, Vec<u64>, u64) {
+        let ibf = ibf_set.get_ibf(ibf_level);
+
+        let mut new_ibf = Ibf::from_vec(ibf_vec.clone(), seed ^ (ibf_level.0 as u64));
+
+        if !new_ibf.merge(&ibf.data, seed ^ (ibf_level.0 as u64)) {
+            error!(target: "network", "exchange routing tables failed with peer {}", peer_id);
+            return (Default::default(), Default::default(), 0);
+        }
+
+        let (edge_hashes, unknown_edges_count) = new_ibf.try_recover();
+        let (known, unknown_edges) = self.split_edges_for_peer(&peer_id, &edge_hashes);
+
+        (known, unknown_edges, unknown_edges_count)
+    }
+}
+#[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+impl Handler<IbfRoutingTableExchangeMessages> for IbfRoutingTableExchangeActor {
+    type Result = IbfRoutingTableExchangeMessagesResponse;
+
+    #[perf]
+    fn handle(
+        &mut self,
+        msg: IbfRoutingTableExchangeMessages,
+        _ctx: &mut Self::Context,
+    ) -> Self::Result {
+        match msg {
+            IbfRoutingTableExchangeMessages::AddEdges(edges) => {
+                for edge in edges.iter() {
+                    let se = edge.to_simple_edge();
+                    self.peer_ibf_set.add_edge(&se);
+                    self.edges.insert((edge.peer0, edge.peer1), edge.clone());
+                }
+            }
+            #[cfg(feature = "adversarial")]
+            IbfRoutingTableExchangeMessages::RemoveEdges(edges) => {
+                for edge in edges.iter() {
+                    self.peer_ibf_set.remove_edge(edge);
+                }
+            }
+            IbfRoutingTableExchangeMessages::RemoveEdges2(edges) => {
+                for edge in edges.iter() {
+                    self.peer_ibf_set.remove_edge(&edge.to_simple_edge());
+                }
+            }
+            IbfRoutingTableExchangeMessages::AddIbfSet(peer_id, ibf_set) => {
+                self.peer_ibf_set.add_peer(peer_id, ibf_set, &mut self.edges);
+                let seed = self.peer_ibf_set.get(&peer_id).unwrap().get_seed();
+                let ibf_vec = self.peer_ibf_set.get(&peer_id).unwrap().get_ibf_vec(MIN_IBF_LEVEL);
```

Comment:
It's better to make a temporary variable out of `self.peer_ibf_set.get(&peer_id).unwrap()`

Reply:
Ok, I removed this unwrap()